### PR TITLE
Modular SpeechLLM implementation for Sept. 2023 submission (SALM)

### PR DESCRIPTION
**Pull request by zhehuaichen**
# What does this PR do ?

Include Modular SpeechLLM implementation for Sept. 2023 submission (SALM). The implementation integrates a pretrained FC and a frozen LLM, and can do ASR, AST, and in-context learning for keyword boosting.

The current implementation is based off of r1.20.0 release.

**Collection**: multimodal

# Usage

refer to `workspace/run_sft_audio_lm.py`

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

**Comment by stevehuang52** [comment]
> @zhehuaichen why merging to https://github.com/NVIDIA/NeMo/tree/modular_speechllm ? instead of main? What is the purpose of files in workspace/* folder?

@nithinraok Because the project is still under development, and we want to put the branch under NVIDIA/NeMo to facilitate collaboration with more people

File: tests/collections/multimodal/test_speechllm_models.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
```
Comment (stevehuang52) [inline tests/collections/multimodal/test_speechllm_models.py:1]: 2023

**Review by stevehuang52 — CHANGES_REQUESTED** [review]
Overall LTGM, but let's remove the `workspace/scripts` and `workspace/tools` folders since the main branch has LLaMA support already.

File: tests/collections/multimodal/test_speechllm_models.py (hunk 1 above)
Comment (zhehuaichen) [reply tests/collections/multimodal/test_speechllm_models.py:1]: done

**Review by stevehuang52 — APPROVED** [review]
LGTM, thanks~

**Pull request merged**